cmd/object-api-utils.go

```diff
@@ -64,12 +69,13 @@ func IsValidBucketName(bucket string) bool {
 	if len(bucket) < 3 || len(bucket) > 63 {
 		return false
 	}
-	if bucket[0] == '.' || bucket[len(bucket)-1] == '.' {
```

Comment:
The above condition would evaluate to false for all valid bucket names. Doesn't it perform regex matching in that case? So, the optimization applies only when the bucket name starts or ends with a '.', which is an invalid bucket name. I understand that we should avoid performing regex matches if we could, but we save nearly nothing in this case.